Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -4861,6 +4922,25 @@ private void ProcessFileChanges(LoggingContext loggingContext, SchedulerState sc
                     m_configuration.Schedule.InputChanges.ToString(Context.PathTable),
                     m_configuration.Layout.SourceDirectory.ToString(Context.PathTable),
                     DirectoryTranslator);
+
+                // The Algorithm can't deal with removal currently, so disable change-based code coverage and have to run full code coverage if has removal. 

```

Comment:
>// The Algorithm can't deal with removal currently, so disable change-based code coverage and have to run full code coverage if has removal [](start = 16, length = 139)

If you have `m_sourceChangeAffectedOutputFiles` and `m_sourceChangeAffectedOutputDirectories` in a separate class, you can use that class to refactor this part of the code.